src/pricer/utils.ts

```diff
@@ -0,0 +1,66 @@
+import type { BaseExchangeApi } from 'src/api';
+import {
+  CoinBase,
+  CoinCodex,
+  CoinGecko,
+  RdocExchange,
+  TestExchange,
+} from '../api';
+
+type ExchangeApiName =
+  | 'coinBase'
+  | 'coinGecko'
+  | 'coinCodex'
+  | 'rdocExchange'
+  | 'testExchange';
+
+const tokenToApi: Record<string, ExchangeApiName[]> = {
+  RBTC: ['coinGecko', 'coinCodex'],
+  RDOC: ['rdocExchange'],
+  RIF: ['coinGecko', 'coinBase', 'coinCodex'],
+  TRIF: ['coinGecko', 'coinBase', 'coinCodex'],
+  TKN: ['testExchange'],
+};
+
+const INTERMEDIATE_CURRENCY = 'USD';
+
+type CoinBaseConstructorArgs = ConstructorParameters<typeof CoinBase>;
+type CoinCodexConstructorArgs = ConstructorParameters<typeof CoinCodex>;
+type CoinGeckoConstructorArgs = ConstructorParameters<typeof CoinGecko>;
+type ConstructorArgs =
+  | CoinBaseConstructorArgs
+  | CoinCodexConstructorArgs
+  | CoinGeckoConstructorArgs;
+
+const builders = new Map<
+  ExchangeApiName,
+  (args?: ConstructorArgs) => BaseExchangeApi
+>();
+builders.set(
+  'coinBase',
+  (args?: CoinBaseConstructorArgs) => new CoinBase(...(args || []))
+);
+builders.set(
+  'coinGecko',
+  (args?: CoinCodexConstructorArgs) => new CoinCodex(...(args || []))
+);
+builders.set(
+  'coinCodex',
+  (args?: CoinGeckoConstructorArgs) => new CoinGecko(...(args || []))
+);
+builders.set('rdocExchange', () => new RdocExchange());
+builders.set('testExchange', () => new TestExchange());
+
+function createExchangeApi(
```

Comment:
actually I'd consider value of this function as it only adds the error on non-existent api, but the only place this is used is in the pricer, which will then error if any one of them does not initialise (guessing that's the idea for the error handling anyway), so will need custom handling anyway. I'd just get it [where it is required](https://github.com/rsksmart/rif-relay-client/pull/85/files#diff-38cd6bf9d92f33682673b200915d827e7231f2bdbc538271a4c719e6cba37bf0R63) directly as `exchangeApis[exchangeApiType]()`